src/zkevm_specs/evm/execution/block_ctx.py

```diff
@@ -0,0 +1,44 @@
+from ...util.param import N_BYTES_ACCOUNT_ADDRESS, N_BYTES_U64
+from ..instruction import Instruction, Transition
+from ..table import BlockContextFieldTag
+from ..opcode import Opcode
+from ...util import FQ
+
+
+def blockctx(instruction: Instruction):
+    opcode = instruction.opcode_lookup(True)
+
+    res_op = instruction.curr.execution_state.responsible_opcode()[0]
+    if isinstance(res_op, int):
+        res_op_int = int(res_op)
+    instruction.constrain_equal(opcode, FQ(res_op_int))
```

Comment:
Is this really needed? In `Instruction.step_state_transition_to_new_context` we're doing `self.responsible_opcode_lookup(opcode)` which I think checks the same thing.
The other opcodes are not doing this constraint check.

Reply:
yeah, nice! when coding i also wonder for `responsible_opcode` and `responsible_opcode_lookup` have double checking for opcodeId.